ts/Series/PointAndFigure/PointAndFigureSeries.ts

```diff
@@ -0,0 +1,341 @@
+            }
+        }
+
+        if (this.isDirtyData || pnfDataGroups.length === 0) {
+
+            this.pnfDataGroups.length = 0;
+
+            // Get first point and determine its symbol and trend
+            for (let i = 0; i < yData.length; i++) {
+                const x = xData[i],
+                    close = yData[i],
+                    firstPoint = yData[0];
+
+                if (close - firstPoint >= calculatedBoxSize) {
+                    upTrend = true;
+                    pnfDataGroups.push({ x, y: [close], upTrend });
+                    break;
+                }
+                if (firstPoint - close >= calculatedBoxSize) {
+                    upTrend = false;
+                    pnfDataGroups.push({ x, y: [close], upTrend });
+                    break;
+                }
+            }
+
+            yData.forEach((close, i): void => {
+                const x = xData[i],
+                    lastPoint = getLastPoint(pnfDataGroups);
+
+                if (upTrend) {
+                    // Add point going UP
+                    if (close - lastPoint >= calculatedBoxSize) {
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+
+                    if (lastPoint - close >= reversal) { // Handle reversal
+                        upTrend = false;
+
+                        pnfDataGroups.push({ x, y: [], upTrend });
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+                }
+
+                if (!upTrend) {
+                    // Add point going DOWN
+                    if (lastPoint - close >= calculatedBoxSize) {
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+
+                    if (close - lastPoint >= reversal) { // Handle reversal
+                        upTrend = true;
+
+                        pnfDataGroups.push({ x, y: [], upTrend });
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+                }
+            });
+        }
+
+        // Process the pnfDataGroups to HC series format
+        const finalData: {x: number, y: number, upTrend: boolean}[] = [];
+
+        series.processedXData.length = 0;
+        series.processedYData.length = 0;
+
+        pnfDataGroups.forEach((point): void => {
+            const x = point.x,
+                upTrend = point.upTrend;
+
+            point.y.forEach((y): void => {
+                series.processedXData.push(x);
+                series.processedYData.push(y as any);
+                finalData.push({
+                    x,
+                    y,
+                    upTrend
+                });
+            });
+        });
+
+        series.pnfDataGroups = pnfDataGroups;
+        series.processedData = finalData;
+    }
+
+    public processData(force?: boolean): (boolean|undefined) {
+        const ret = super.processData(force);
+
+        if (this.is?.('pointandfigure')) {
```

Comment:
you don't need to check if it is point and figure. you already are in that function

Reply:
This check is intentional, to since this function is called more than once, not only for PNF series (I think it was for navigator, but I'd have to check again)